### PR TITLE
Check for Tape errors before submitting a job

### PR DESCRIPTION
**Pull request by shabtaisharon**
Before submitting a job check that there are no tape errors.  If there is, error out letting the user know. 
If the --force flag is specified, go ahead and submit the request anyway.

### CODE REVIEW

File: ds3-cli-integration/src/main/java/com/spectralogic/ds3cli/integration/Util.java [hunk 1]
```diff
@@ -32,7 +32,8 @@
     public static final String RESOURCE_BASE_NAME = "./src/test/resources/books/";
     public static final String DOWNLOAD_BASE_NAME = "./output/";
 
-    private Util() {}
+    private Util() {
```
Comment (DenverM80) [inline ds3-cli-integration/src/main/java/com/spectralogic/ds3cli/integration/Util.java]: unnecessary whitespace change?

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetBulk.java [hunk 2]
```diff
@@ -149,27 +154,26 @@ public Ds3Object apply(final Contents input) {
...
+                        .withPriority(this.priority));
 
         job.transfer(new LoggingFileObjectGetter(getter));
 
         if (sync) {
             if (prefix != null) {
                 return "SUCCESS: Synced all the objects that start with '" + this.prefix + "' from " + this.bucketName + " to " + this.outputPath.toString();
-            }
-            else {
+            } else {
```
Comment (DenverM80) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetBulk.java]: nice.  bonus points for fixing these formatting inconsistencies.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetBulk.java [hunk 3]
```diff
@@ -149,27 +154,26 @@ public Ds3Object apply(final Contents input) {
...
-        return "SUCCESS: Wrote all the objects that start with '"+ this.prefix +"' from " + this.bucketName + " to " + this.outputPath.toString();
+        return "SUCCESS: Wrote all the objects that start with '" + this.prefix + "' from " + this.bucketName + " to " + this.outputPath.toString();
     }
 
     private String restoreAll(final Ds3ClientHelpers.ObjectChannelBuilder getter) throws XmlProcessingException, SignatureException, IOException, SSLSetupException {
         final Ds3ClientHelpers helper = getClientHelpers();
         final Ds3ClientHelpers.Job job = helper.startReadAllJob(this.bucketName,
                 ReadJobOptions.create()
-                .withPriority(this.priority));
+                        .withPriority(this.priority));
```
Comment (DenverM80) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetBulk.java]: I think this indentation is extra - double indent (8 spaces) indicates continuation from the previous line, this is triple indented.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/PutBulk.java [hunk 4]
```diff
@@ -123,8 +129,8 @@ public PutBulkResult call() throws Exception {
 
         final Ds3ClientHelpers.Job job = helpers.startWriteJob(this.bucketName, ds3Objects,
                 WriteJobOptions.create()
-                .withPriority(this.priority)
-                .withWriteOptimization(this.writeOptimization));
+                        .withPriority(this.priority)
```
Comment (DenverM80) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/PutBulk.java]: same extra indentation here

**Comment by shabtaisharon** [comment]
saw the failed checks, i'm on it

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/BlackPearlUtils.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import com.spectralogic.ds3client.commands.GetTapeFailureRequest;
+import com.spectralogic.ds3client.models.tape.TapeFailure;
+
+import java.io.IOException;
+import java.security.SignatureException;
+
+public final class BlackPearlUtils {
+
+    public static void checkBlackPearlForTapeFailure(final Ds3Client client) throws IOException, SignatureException, TapeFailureException {
+        final Iterable<TapeFailure> tapeFailures = client.getTapeFailure(new GetTapeFailureRequest()).getTapeFailures();
```
Comment (DenverM80) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/BlackPearlUtils.java]: I don't see where client.getTapeFailure() is implemented - was it already done and you're just calling it here?

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetBucket.java [hunk 6]
```diff
@@ -17,7 +17,7 @@
 
 
 import com.spectralogic.ds3cli.Arguments;
-import com.spectralogic.ds3cli.CommandException;
+import com.spectralogic.ds3cli.Exceptions.CommandException;
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetBucket.java]: packages should be lowercase

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/PutBulk.java (hunk 4 above)
Comment (shabtaisharon) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/PutBulk.java]: this is the suggested indentation the IDE gives me

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/BlackPearlUtils.java (hunk 5 above)
Comment (shabtaisharon) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/BlackPearlUtils.java]: this is in the java dsk

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/DeleteBucket.java [hunk 7]
```diff
@@ -64,13 +65,17 @@ public DeleteResult call() throws Exception {
...
 
-    private String deleteBucket() throws SignatureException, SSLSetupException, CommandException {
+    private String deleteBucket() throws SignatureException, SSLSetupException, CommandException, IOException {
         try {
             getClient().deleteBucket(new DeleteBucketRequest(bucketName));
         }
-        catch (final IOException e) {
+        catch (final FailedRequestException e) {
+            if (e.getStatusCode() == 409) { //BUCKET_NOT_EMPTY
+                return "Error: Tried to delete a non-empty bucket without the force delete objects flag.\nUse --force to delete all objects in the bucket";
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/DeleteBucket.java]: This should return a command exception.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetObject.java [hunk 8]
```diff
@@ -88,34 +96,30 @@ public GetObjectResult call() throws Exception {
...
                 }
             }
 
             Transfer(helpers, ds3Obj);
             return new GetObjectResult("SUCCESS: Finished downloading object.  The object was written to: " + filePath);
-        }
-        catch(final FailedRequestException e) {
-            if(e.getStatusCode() == 500) {
+        } catch (final FailedRequestException e) {
+            if (e.getStatusCode() == 500) {
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetObject.java]: It might be cleaner to use a switch statement here.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetBucketView.java [hunk 9]
```diff
@@ -62,7 +63,7 @@ public String render(final GetBucketResult br) {
                 new ASCIITableHeader("ETag", ASCIITable.ALIGN_RIGHT)};
     }
 
-    private String nullGuard(String message) {
+    private String nullGuard(final String message) {
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetBucketView.java]: With the v1.2.3 sdk this can be updated to use the Guard class.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Exceptions/TapeFailureException.java [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+            arrayEntry[4] = nullGuard(tapeFailure.getType().toString());
+            arrayEntry[5] = nullGuard(tapeFailure.getErrorMessage());
+
+            contents.add(arrayEntry);
+        }
+
+        return contents.toArray(new String[contents.size()][]);
+    }
+
+    private static String nullGuard(final String message) {
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Exceptions/TapeFailureException.java]: This function is in some of the other tape printing places, we should probably consolidate this into a single util class.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetBucketView.java (hunk 9 above)
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetBucketView.java]: Same comment as above about consolidating this into a single table view util function

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetObject.java [hunk 11]
```diff
@@ -88,34 +96,31 @@ public GetObjectResult call() throws Exception {
...
-                if (LOG.isInfoEnabled()) {
-                    e.printStackTrace();
-                }
-                throw new CommandException( "Error: Encountered an unknown error of ("+ e.getStatusCode() +") while accessing the remote DS3 appliance.", e);
+        } catch (final FailedRequestException e) {
+            switch (e.getStatusCode()) {
+                case 500:
+                    LOG.info(e.getMessage());
+                    if (LOG.isInfoEnabled()) {
+                        e.printStackTrace();
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetObject.java]: Don't print the stack trace.  The logging code will print it once it gets caught later.
